Library/Homebrew/dev-cmd/pull.rb

```diff
@@ -145,29 +145,34 @@ def pull
         opoo "Current branch is #{branch}: do you need to pull inside master?"
       end
 
-      patch_puller = PatchPuller.new(url, args)
-      patch_puller.fetch_patch
-      patch_changes = files_changed_in_patch(patch_puller.patchpath, tap)
-
-      is_bumpable = patch_changes[:formulae].length == 1 && patch_changes[:others].empty?
-      if do_bump
-        odie "No changed formulae found to bump" if patch_changes[:formulae].empty?
-        if patch_changes[:formulae].length > 1
-          odie "Can only bump one changed formula; bumped #{patch_changes[:formulae]}"
+      merge_commit = merge_commit?(url)
+
+      unless merge_commit
+        patch_puller = PatchPuller.new(url, args)
+        patch_puller.fetch_patch
+        patch_changes = files_changed_in_patch(patch_puller.patchpath, tap)
+
+        is_bumpable = patch_changes[:formulae].length == 1 && patch_changes[:others].empty?
+        if do_bump
+          odie "No changed formulae found to bump" if patch_changes[:formulae].empty?
+          if patch_changes[:formulae].length > 1
+            odie "Can only bump one changed formula; bumped #{patch_changes[:formulae]}"
+          end
+          odie "Can not bump if non-formula files are changed" unless patch_changes[:others].empty?
         end
-        odie "Can not bump if non-formula files are changed" unless patch_changes[:others].empty?
-      end
-      if is_bumpable
-        old_versions = current_versions_from_info_external(patch_changes[:formulae].first)
+        if is_bumpable
+          old_versions = current_versions_from_info_external(patch_changes[:formulae].first)
+        end
+        patch_puller.apply_patch
       end
-      patch_puller.apply_patch
 
       changed_formulae_names = []
 
+      end_revision = merge_commit ? "FETCH_HEAD" : "HEAD"
```

Comment:
I'm not crazy on relying on using `FETCH_HEAD` head. I consider it to be transient, since the next `git fetch` would blow it away. It could create a difficult to troubleshoot bug if some function called some function that called `git fetch`. I suggest checking out the result of `FETCH_HEAD` to a named branch to update `HEAD`, and then this logic can go away.

Reply:
I think it's fine to use `FETCH_HEAD` as long as we're doing the `git fetch` immediately before getting its value. Creating a new branch should be unnecessary but `FETCH_HEAD` should be stored in a variable right after the fetch.

Reply:
How about we store the SHA1 in that variable rather than literal `FETCH_HEAD`?

Reply:
I changed the code to do that.

Reply:
Works for me 👍